src/main/java/org/junit/runners/BlockJUnit4ClassRunner.java

```diff
@@ -379,12 +382,13 @@ private Statement withRules(FrameworkMethod method, Object target,
 
     private Statement withMethodRules(FrameworkMethod method, List<TestRule> testRules,
             Object target, Statement result) {
+        Statement withMethodRules = result;
         for (org.junit.rules.MethodRule each : getMethodRules(target)) {
-            if (!testRules.contains(each)) {
-                result = each.apply(result, method, target);
+            if (!(each instanceof TestRule && testRules.contains(each))) {
```

Comment:
More clear way to specify why we are looking for the `MethodRule` in list of  `TestRule ` isn't?